### PR TITLE
Feature: Union schema compatibility

### PR DESCRIPTION
**Pull request by fivetran-catfritz**
Issue: #18 

Confirm the following files were correctly updated automatically:
- [x] CHANGELOG
- [x] README
- [x] stg_amazon_ads.yml, src_amazon_ads.yml, amazon_ads.yml (depending if source or transform)
- [x] docs.md
- [ ] ~joins~
- [x] window functions (partition by)
- [x] 'source_relation' column added in staging and transform models
- [x] tests

Manual updates:
- [ ] ~Add `source_relation` to downstream models if necessary~
- [x] Finish any incomplete/incorrect joins/partitions
- [x] Update tests to include `source_relation` in unique-combination-of-cols if necessary
    - May need to remove some uniqueness tests in favor of the combo test
    
Validation:
- [x] `dbt run` locally passes
- [x] `dbt test` locally passes

### CODE REVIEW

File: packages.yml
Comment (fivetran-joemarkiewicz) [inline packages.yml:6]: Reminder to update before merge

**Review by fivetran-joemarkiewicz — APPROVED** [review]
These changes look good to go once the fivetran_utils dependency is updated and we can coordinate rolling all of these out together.

File: CHANGELOG.md
Comment (fivetran-joemarkiewicz) [inline CHANGELOG.md:4]: Actually, can we remove the README "hide" sections in these updates. This way we are able to link directly to the section in the README where the users needs to understand how to configure the union feature.

Otherwise, I worry this cool feature is hidden from people actually using it.

Comment (fivetran-catfritz) [reply CHANGELOG.md]: Updated.

**Comment by fivetran-catfritz** [comment]
Merging for release on 12-Oct-2023.

**Pull request merged**
